### PR TITLE
Update analytics.net package reference

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Update analytics.net package reference for new analytics.net build consuming latest .NET6 [pre-release build](https://wiki.autodesk.com/pages/viewpage.action?spaceKey=ADP&title=ADP+-+Elm+Desktop+SDK+v4.101.1+%28ML-Eval%29+-+Release+Notes) of ADP SDK

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCore/DynamoCore.csproj [hunk 1]
```diff
@@ -158,10 +149,6 @@
     <Copy SourceFiles="$(ProjectDir)\BuiltInAndOperators\Operators.xml" DestinationFolder="$(OutputPath)\$(UICulture)" />
     <Copy SourceFiles="$(ProjectDir)\BuiltInAndOperators\BuiltIn.xml" DestinationFolder="$(OutputPath)\$(UICulture)" />
     <Copy SourceFiles="$(ProjectDir)\BuiltInAndOperators\BuiltIn.Migrations.xml" DestinationFolder="$(OutputPath)" />
-    <!--Enable these if/when we need to keep old binaries for patch installer compatability-->
-    <Copy SourceFiles="@(LegacyBinariesToBinFolder)" DestinationFolder="$(OutputPath)" Condition=" '$(TargetFramework)' == 'net48' " />
-    <Copy SourceFiles="@(LegacyBinariesToNodesFolder)" DestinationFolder="$(OutputPath)\nodes\" Condition=" '$(TargetFramework)' == 'net48' " />
-     <Copy SourceFiles="@(LegacyBinariesToBinResourcesFolder)" DestinationFolder="$(OutputPath)\en-US\" Condition=" '$(TargetFramework)' == 'net48' " />
```
Comment (QilongTang) [inline src/DynamoCore/DynamoCore.csproj:164]: hmm Why are these part of this PR? If we want to remove this step, do we want to remove the corresponding binaries from the legacy folder or even the folder itself?

File: test/DynamoCoreTests/DynamoCoreTests.csproj [hunk 2]
```diff
@@ -149,18 +133,6 @@
     <Copy SourceFiles="$(PkgMicrosoft_Diagnostics_Runtime)\lib\netstandard2.0\Microsoft.Diagnostics.Runtime.dll" DestinationFolder="$(TestDependenciesPath)" />
 </Target>
 
-  <Target Name="MoveSomeTestDepsOutOfBin_net48" AfterTargets="Build" Condition=" '$(TargetFramework)' == 'net48' ">
```
Comment (QilongTang) [inline test/DynamoCoreTests/DynamoCoreTests.csproj:152]: I am not 100% sure about the purpose of this.. @pinzart90 and you would know better than me 👍🏻

**Review by QilongTang — COMMENTED** [review]
LGTM with one comment

File: src/DynamoCore/DynamoCore.csproj (hunk 1 above)
Comment (aparajit-pratap) [reply src/DynamoCore/DynamoCore.csproj:164]: I can remove the legacy folder as well.

Comment (aparajit-pratap) [reply src/DynamoCore/DynamoCore.csproj:164]: This strictly doesn't need to belong to this PR but I thought why not clean up this file as I was updating it anyway.

File: test/DynamoCoreTests/DynamoCoreTests.csproj (hunk 2 above)
Comment (aparajit-pratap) [reply test/DynamoCoreTests/DynamoCoreTests.csproj:152]: The reason I removed it is as we are not targeting net48 in this branch anymore.

**Comment by sm6srw** [comment]
I am OK with removing the legacy folder now.

**Review by sm6srw — APPROVED** [review]

File: src/DynamoCore/DynamoCore.csproj [hunk 3]
```diff
@@ -39,7 +35,7 @@
 	<PackageReference Include="Lucene.Net" Version="4.8.0-beta00016" />
     <PackageReference Include="Lucene.Net.Analysis.Common" Version="4.8.0-beta00016" />
     <PackageReference Include="Lucene.Net.QueryParser" Version="4.8.0-beta00016" />
-    <PackageReference Include="DynamoVisualProgramming.Analytics" Version="4.0.0.1432" />
+    <PackageReference Include="DynamoVisualProgramming.Analytics" Version="4.1.0.2298" />
```
Comment (mjkkirschner) [inline src/DynamoCore/DynamoCore.csproj:38]: is this something you just published?

**Review by mjkkirschner — APPROVED** [review]

File: src/DynamoCore/DynamoCore.csproj (hunk 3 above)
Comment (aparajit-pratap) [reply src/DynamoCore/DynamoCore.csproj:38]: Yes

**Review by QilongTang — APPROVED** [review]

**Pull request merged**